### PR TITLE
Dr doc qs bug

### LINKED ISSUE #300: Delayed tasks that are paused/resumed are added again to delayed_tasks array
when you pause and then resume a task, it sets the task's status to "started" by default. however, if the task was delayed before the user clicked pause, while it correctly detects the delayed status, it adds the task to the delayed_tasks array again, which causes that task to be in the delayed_tasks array many times.

Should probably do a check using isDelayed in the resumeTask function (all in awareness.js):
image

### PR DESCRIPTION
**Pull request by dretelny**
This PR attempts to fix the issue preventing users from sometimes completing the task. Specifically, this PR aims to fix issues related to updating the answers to documentation questions in realtime in all views. It also fixes some minor bugs I encountered as I was coding. 

1) In author mode (e.g., before the team has started), the team now updates properly in all views. Specifically, if the author invites workers while still authoring the team, the workers will see the changes as they occur (previously they would only see the event blocks but the event information / members assigned would not show up). This also solves the issue related to certain event information (e.g., if a new member gets assigned) not updating in all views in edit mode (you would have to refresh the page or wait for another event to get added, which would trigger the update)

2) Previously, if you would answer documentation questions and then click on the "Save and Close" button, the flashTeamsUpdated function would recognize that the event was in fact updated so it wouldn't trigger the update. If another view called updateStatus or you refreshed the page, the answers for uncompleted events would be overwritten. 

3) I'm hoping that the fix described in item 2 above will also fix the bug preventing some events from completing (e.g., a worker would answer the questions, click complete and then the event would complete for a second and then quickly switch back to in progress and erase their answers to the documentation questions). Since this bug is quite difficult to replicate for a variety of reasons, I decided to save the worker's answers to the documentation questions in realtime as the worker types -- as shown in the screenshot below (you can see the work in progress answers in the modal on the left): 

image

While this may be inefficient (e.g., whenever a worker is typing, updateStatus is constantly getting called), it seems better than risking that their answers get erased if the bug occurs. Having said that, it is important that we test this to make sure that constantly updating doesn't cause problems in other views or conflicts (I don't think it should but just double check). Also, note that this solution makes the "Cancel" button and the "X" icon on the top right meaningless since canceling/closing the modal is technically the same as the save and close button since the answers are saved in realtime. 

4) Paused tasks now edit properly in edit mode. Previously, while you couldn't edit in progress tasks, you could edit paused tasks. However, if you changed the duration of a paused task, the timer would not update properly and the event would get messed up. When testing this, it is important to pause an event that is in progress and then shorten the duration such that the event would now be delayed and make sure the task turns red when resumed and that the timer is correct. Similarly, try pausing a delayed task and then adding time so it is no longer delayed. Make sure the timer updates correctly and the task turns blue (instead of red). 

5) I also fixed #300 (quick summary: when a paused task that was previously delayed was resumed, it would get added to the delayed_tasks array again causing it to be in the array multiple times). 

When testing: 
1) Create a team and add at least two members (DONT START THE TEAM YET)
2) Invite the two members and open Foundry (ideally in separate browsers if possible) -- you should have three views of Foundry open (1 author view and 2 member views)
3) Add events to the workflow and make sure they properly appear in the member view (for some of the events, add some outputs so we can test the output documentation questions)
4) Try editing some of the details of an existing event (e.g., add or remove a member assigned to the task) and then make sure that the update automatically appears in the worker views
5) Start the team
6) In one of the worker views, start a task. Wait a little bit (like 10 seconds) and then click on complete and start answering the documentation questions (make sure to test this with the general task questions and the output questions)
7) As you are typing your answers in one view (e.g., don't click save or complete), open the task modal of that event in one of the other views and make sure you can see the latest answers
8) Click on the save and close button and then check that the documentation answers properly appear in all views 
9) Start another task, wait ~10 seconds and then open the documentation question modal
10) In the other views, start and pause a few tasks (while the documentation question modal is open in the other window)
11) To test the fixes to editing a paused event in runtime, pause an event, enter edit mode and then change the timer (test this for delayed and not delayed tasks as described in change number 4 in the list of changes above)
12) Go back to the other window and answer the documentation questions and complete the task -- make sure the task properly completes 

That should be all! As always, feel free to test other edge cases I may have overlooked.

### CODE REVIEW

**Comment by mbernst** [comment]
I can't tell exactly when completeTaskModalText gets called offhand --- are you saving with every keystroke, or on blur() events for the various inputs in the form?

File: app/assets/javascripts/authoring/taskStatus.js [hunk 1]
```diff
@@ -328,7 +331,7 @@ function completeTaskModalText(eventToComplete) {
             questions = outputFilledQ[eventOutputs[i]];
             for (j = 0; j < questions.length; j++){
                 if (questions[j] != ""){
-                    modalText += '<p id="textoutput' + i + 'q' + j + '">' + questions[j][0] + '</p></br><textarea id = "output' + i + 'q' + j + '" class="outputForm" rows="3" onkeyup="keyUpFunc()">' + questions[j][1] + '</textarea></br>';
+                    modalText += '<p id="textoutput' + i + 'q' + j + '">' + questions[j][0] + '</p></br><textarea id = "output' + i + 'q' + j + '" class="outputForm" rows="3" onkeyup="keyUpFunc(); saveDocQuestions(' + eventToComplete.id + ')">' + questions[j][1] + '</textarea></br>';
```
Comment (dretelny) [inline app/assets/javascripts/authoring/taskStatus.js]: @mbernst this is one of the two places where the documentation answers are saved (on keyup)

File: app/assets/javascripts/authoring/taskStatus.js [hunk 2]
```diff
@@ -345,7 +348,7 @@ function completeTaskModalText(eventToComplete) {
         else{
             var placeholderVal = generalFilledQ[i][1]; 
         }
-        modalText += '<p id="textq' + i + '">' + generalQuestions[i] + ': </p></br><textarea id="q' + i + '"class="outputForm" rows="3" onkeyup="keyUpFunc()">'+ placeholderVal + '</textarea></br>';
+        modalText += '<p id="textq' + i + '">' + generalQuestions[i] + ': </p></br><textarea id="q' + i + '"class="outputForm" rows="3" onkeyup="keyUpFunc(); saveDocQuestions(' + eventToComplete.id + ')">'+ placeholderVal + '</textarea></br>';
```
Comment (dretelny) [inline app/assets/javascripts/authoring/taskStatus.js]: @mbernst this is the other place where the documentation answers are saved (on keyup)
